### PR TITLE
movie: Add clock init time to CTM header

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
This adds a clock init time field to the CTM header. The clock settings would be overridden when playing a movie. And when recording a movie, if the clock is set to System Time, it would be set to fixed init time at the current moment as well. In this way this keeps consistency with the RNG even if the user does just no setting.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4267)

### CODE REVIEW

**Comment by InfamousKnight** [comment]
Nice! Has frame advance been added yet? Could be useful for segmented runs while save states are unavailable.

File: src/citra/citra.cpp [hunk 1]
```diff
@@ -332,7 +339,7 @@ int main(int argc, char** argv) {
     }
 
     if (!movie_play.empty()) {
-        Core::Movie::GetInstance().StartPlayback(movie_play);
+        Core::Movie::GetInstance().StartPlayback(movie_play, [] {});
```
Comment (wwylele) [inline src/citra/citra.cpp]: You should change the default param for `StartPlayback` or its internal logic if the current default isn't a meaningful value.

Comment (zhaowenlan1779) [reply src/citra/citra.cpp]: @wwylele fixed

**Comment by zhaowenlan1779** [comment]
I'll merge this in 6 hours if no more comments.

**Pull request merged**